syntax = "proto3";

package Node;

service Node {
    rpc requestVote (voteRequest) returns (voteResponse);
    rpc appendEntries (appendEntriesRequest) returns (appendEntriesResponse);
}

message voteRequest {
    int64 term = 1;
    int64 candidate_id = 2;
    int64 last_log_index = 3;
    int64 last_log_term = 4;
}

message voteResponse {
    int64 term = 1;
    bool granted = 2;
    int64 node_id = 3;
}

message appendEntriesRequest {

}

message appendEntriesResponse {
    
}